src/box/relay.cc

```diff
@@ -422,11 +426,26 @@ relay_set_cord_name(int fd)
 	cord_set_name(name);
 }
 
+/** A callback invoked on xrow_buf flush pushing its contents to the peer. */
+static int
+relay_flush_cb(void *data, struct iovec *iov, int iovcnt)
+{
+	struct iostream *io = (struct iostream *)data;
+	if (coio_writev(io, iov, iovcnt, 0) < 0)
+		return -1;
+
+	struct errinj *inj = errinj(ERRINJ_RELAY_TIMEOUT, ERRINJ_DOUBLE);
+	if (inj != NULL && inj->dparam > 0)
+		fiber_sleep(inj->dparam);
```

Comment:
Why not leave it in `relay_buffer_row` (former `relay_send`)? Then we wouldn't need to update the tests AFAICS.

Reply:
It's not possible, unfortunately. We have to update the tests anyway.

Some tests (`replication/sync`, for example) rely on the fact that upstream lag will become tiny eventually, even if relay is slow to send rows. This wouldn't be possible if we started waiting **after** the `send`, but **before** the `flush`.

Speaking of altering the batch size in some tests, they rely on the fact that relay performs a sequence of sends with some given timeout in between. It's just easier to perform the needed amount of sends with a tiny batch size. So that I don't have to create ~16 kilobytes of data per each desired send.